### PR TITLE
[FLINK-33164] Support write option sink.ignore-null-value

### PR DESCRIPTION
**Pull request by Tan-JiaLiang**
## What is the purpose of the change

Sometimes, user want to write data and ignore null value to achieve partial update. 

## Brief change log

  - *support write option sink.ignore-null-value*

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

This change is already covered by existing tests, such as
  - *org.apache.flink.connector.hbase1.HBaseDynamicTableFactoryTest#testSinkIgnoreNullValueOptions*
  - *org.apache.flink.connector.hbase2.HBaseDynamicTableFactoryTest#testSinkIgnoreNullValueOptions*
  - *org.apache.flink.connector.hbase.util.HBaseSerdeTest#writeIgnoreNullValueTest*

### CODE REVIEW

File: flink-connector-hbase-1.4/src/main/java/org/apache/flink/connector/hbase1/HBase1DynamicTableFactory.java
Comment (ferenc-csaky) [inline flink-connector-hbase-1.4/src/main/java/org/apache/flink/connector/hbase1/HBase1DynamicTableFactory.java]: I suggest to avoid wildard imports.

File: flink-connector-hbase-1.4/src/test/java/org/apache/flink/connector/hbase1/HBaseDynamicTableFactoryTest.java
Comment (ferenc-csaky) [inline flink-connector-hbase-1.4/src/test/java/org/apache/flink/connector/hbase1/HBaseDynamicTableFactoryTest.java]: I think instantiating a dummy expected object can be avoided, we can simply use `assertThat(actual.isIgnoreNullValue()).isTrue()`.

File: flink-connector-hbase-2.2/src/main/java/org/apache/flink/connector/hbase2/HBase2DynamicTableFactory.java
Comment (ferenc-csaky) [inline flink-connector-hbase-2.2/src/main/java/org/apache/flink/connector/hbase2/HBase2DynamicTableFactory.java]: I suggest to avoid wildard imports.

File: flink-connector-hbase-2.2/src/test/java/org/apache/flink/connector/hbase2/HBaseDynamicTableFactoryTest.java
Comment (ferenc-csaky) [inline flink-connector-hbase-2.2/src/test/java/org/apache/flink/connector/hbase2/HBaseDynamicTableFactoryTest.java]: I think instantiating a dummy expected object can be avoided, we can simply use `assertThat(actual.isIgnoreNullValue()).isTrue()`.

File: flink-connector-hbase-base/src/main/java/org/apache/flink/connector/hbase/table/HBaseConnectorOptionsUtil.java
Comment (ferenc-csaky) [inline flink-connector-hbase-base/src/main/java/org/apache/flink/connector/hbase/table/HBaseConnectorOptionsUtil.java]: I suggest to avoid wildard imports.

File: flink-connector-hbase-base/src/test/java/org/apache/flink/connector/hbase/util/HBaseSerdeTest.java
Comment (ferenc-csaky) [inline flink-connector-hbase-base/src/test/java/org/apache/flink/connector/hbase/util/HBaseSerdeTest.java]: Pls. use `Assert4J`'s `assertThat` for all kinds of asserts. The community agreed that it should be the standard going forward as part of the [JUnit5 migration](https://docs.google.com/document/d/1514Wa_aNB9bJUen4xm5uiuXOooOJTtXqS_Jqk9KJitU/edit#heading=h.n2pt1kpwyals). It is not applied completely to this repo yet, but IMO it makes sense to conform with that if we touch some tests.

**Review by ferenc-csaky — CHANGES_REQUESTED** [review]
Thank you for your contribution! The proposal and the logic makes sense IMO. Added some comments.

**Comment by Tan-JiaLiang** [comment]
@ferenc-csaky Thanks for the code review, very helpful suggestions. I have changed it, please have a look.

**Review by ferenc-csaky — APPROVED** [review]
LGTM!

**Comment by ferenc-csaky** [comment]
I would cherry-pick the changes relevant of this PR against the current `main` into a new local branch and force push that  to this origin branch, cause right now it contains changes and commits it should not.

**Comment by Tan-JiaLiang** [comment]
@ferenc-csaky I rebase the main branch and squash all my changes.

And one more question master. there will be multiple commits during code reviews, Should I squash all commit into one and apply force push to the origin? Or keep it and wait for the maintainer to squash and merge?

**Comment by ferenc-csaky** [comment]
> @ferenc-csaky I rebase the main branch and squash all my changes.
> 
> And one more question master. there will be multiple commits during code reviews, Should I squash all commit into one and apply force push to the origin? Or keep it and wait for the maintainer to squash and merge?

I think you can follow whichever path is more comfortable for you and for the specific case at hand. If there are multiple commits in the PR the maintainer will squash it, no problem.

@MartijnVisser any thoughts on this?

**Review by ferenc-csaky — APPROVED** [review]
LGTM!

**Comment by MartijnVisser** [comment]
It's recommended to have multiple commits in a lot of situations, see https://flink.apache.org/how-to-contribute/code-style-and-quality-pull-requests/ - We can squash before merging

**Comment by Tan-JiaLiang** [comment]
@ferenc-csaky @MartijnVisser Thanks for the patience guide. Making sense to me.

**Comment by ferenc-csaky** [comment]
@MartijnVisser this is also in a mergeable state I think.

**Review by MartijnVisser — APPROVED** [review]

**Pull request merged**
